src/cryptenroll/cryptenroll.c

```diff
@@ -486,6 +495,30 @@ static int prepare_luks(
                                 strlen(envpw));
                 if (r < 0)
                         return log_error_errno(r, "Password from environment variable $PASSWORD did not work.");
+        } else if(arg_unlock_file) {
+                char *password;
```

Comment:
`_cleanup_(erase_and_freep) char *password = NULL`

Reply:
still not addressed

Reply:
changing this gives me:
```
```

Reply:
bah, it looks like it's not an actual string, it's a hidden special structure. You'll need to add a build time check for crypt_safe_free, as it's available in newer versions than the minimum requirement, and wrap the whole block in an ifdef and use crypt_safe_free

Reply:
dont, just use read_full_file() see below, and then use _cleanup_(erase_and_freep)

Reply:
also, given this is now binary (i.e. not a NUL-terminated string anymore), please use `void*`, and not `char*` (which suggests this was a string)